{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run pipeline with bash (You will need to install salmon, fastQC and multiQC first)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/jespinosa/git/BovReg-Reproducibility/rnaseq-nf'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os, shutil\n",
    "pwd = os.getcwd()\n",
    "os.getcwd()\n",
    "pipeline_folder = \"rnaseq-nf\" \n",
    "work_dir = pwd + \"/../\" + pipeline_folder\n",
    "os.chdir(work_dir)\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "[2020-01-17 16:47:28.586] [jLog] [info] building index\n",
      "out : index\n",
      "\u001b[00m[2020-01-17 16:47:28.586] [puff::index::jointLog] [info] Running fixFasta\n",
      "\u001b[00m\n",
      "[Step 1 of 4] : counting k-mers\n",
      "\n",
      "\u001b[00m[2020-01-17 16:47:28.590] [puff::index::jointLog] [info] Replaced 0 non-ATCG nucleotides\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.590] [puff::index::jointLog] [info] Clipped poly-A tails from 0 transcripts\n",
      "\u001b[00mwrote 1 cleaned references\n",
      "seqHash 256 : c7b06752fa0748ecfa05ab97e8460b69c408b6d5fedf9de3d0d53b05570cf942\n",
      "seqHash 512 : ed859a805ddc0e6805be16dea7160060c1ecf1853f01fe24e64779390dd0b4aa157dab4d3e6234e9f7a1a80cd74df05454259b325624f601a0f86ea508550a8e\n",
      "nameHash 256 : 6b86b273ff34fce19d6b804eff5a3f5747ada4eaa22f1d49c01e52ddb7875b4b\n",
      "nameHash 512 : 4dff4ea340f0a823f15d3f4f01ab62eae0e5da579ccb851f8db9dfe84c58b2b37b89903a740e1ee172da793a6e79d560e5f7f9bd058a12a280433ed6fa46510a\n",
      "\u001b[00m[2020-01-17 16:47:28.590] [puff::index::jointLog] [info] Filter size not provided; estimating from number of distinct k-mers\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.592] [puff::index::jointLog] [info] ntHll estimated 147846 distinct k-mers, setting filter size to 2^22\n",
      "\u001b[00mThreads = 1\n",
      "Vertex length = 31\n",
      "Hash functions = 5\n",
      "Filter size = 4194304\n",
      "Capacity = 2\n",
      "Files: \n",
      "index/ref_k31_fixed.fa\n",
      "--------------------------------------------------------------------------------\n",
      "Round 0, 0:4194304\n",
      "Pass\tFilling\tFiltering\n",
      "1\t0\t0\t\n",
      "2\t0\t0\n",
      "True junctions count = 7\n",
      "False junctions count = 236\n",
      "Hash table size = 243\n",
      "Candidate marks count = 250\n",
      "--------------------------------------------------------------------------------\n",
      "Reallocating bifurcations time: 0\n",
      "True marks count: 16\n",
      "Edges construction time: 0\n",
      "--------------------------------------------------------------------------------\n",
      "Distinct junctions = 7\n",
      "\n",
      "approximateContigTotalLength: 92130\n",
      "counters:\n",
      "1 0 0 0\n",
      "contig count: 14 element count: 171333 complex nodes: 1\n",
      "size: 171333\n",
      "# of ones in rank vector: 13\n",
      "size: 171333\n",
      "\u001b[00m[2020-01-17 16:47:28.678] [puff::index::jointLog] [info] Setting the index/BinaryGfa directory index\n",
      "\u001b[00msize = 171333\n",
      "-----------------------------------------\n",
      "| Loading contigs | Time = 20.268 us\n",
      "-----------------------------------------\n",
      "size = 171333\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 10.715 us\n",
      "-----------------------------------------\n",
      "Number of ones: 13\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 1\n",
      "\u001b[00m[2020-01-17 16:47:28.678] [puff::index::jointLog] [info] Done wrapping the rank vector with a rank9sel structure.\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.678] [puff::index::jointLog] [info] contig count for validation: 13\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.678] [puff::index::jointLog] [info] Total # of Contigs : 13\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.678] [puff::index::jointLog] [info] Total # of numerical Contigs : 13\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.678] [puff::index::jointLog] [info] \n",
      "Total # of segments we have position for : 13\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.679] [puff::index::jointLog] [info] total contig vec entries 17\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.679] [puff::index::jointLog] [info] bits per offset entry 5\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.679] [puff::index::jointLog] [info] there were 1  equivalence classes\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.679] [puff::index::jointLog] [info] # segments = 13\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.679] [puff::index::jointLog] [info] total length = 171,333\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.679] [puff::index::jointLog] [info] Reading the reference files ...\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.680] [puff::index::jointLog] [info] positional integer width = 18\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.680] [puff::index::jointLog] [info] seqSize = 171,333\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.680] [puff::index::jointLog] [info] rankSize = 171,333\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.680] [puff::index::jointLog] [info] edgeVecSize = 0\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.680] [puff::index::jointLog] [info] num keys = 170,943\n",
      "\u001b[00mfor info, total work write each  : 2.331    total work inram from level 3 : 4.322  total work raw : 25.000 \n",
      "[Building BooPHF]  100  %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec\n",
      "Bitarray          901696  bits (100.00 %)   (array + ranks )\n",
      "final hash             0  bits (0.00 %) (nb in final hash 0)\n",
      "\u001b[00m[2020-01-17 16:47:28.693] [puff::index::jointLog] [info] mphf size = 0.107491 MB\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.693] [puff::index::jointLog] [info] chunk size = 171,333\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.693] [puff::index::jointLog] [info] chunk 0 = [0, 171,303)\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.701] [puff::index::jointLog] [info] finished populating pos vector\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.701] [puff::index::jointLog] [info] writing index components\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:28.701] [puff::index::jointLog] [info] finished writing dense pufferfish index\n",
      "\u001b[00m[2020-01-17 16:47:28.701] [jLog] [info] done building index\n"
     ]
    }
   ],
   "source": [
    "!salmon index --threads 1 -t ./data/ggal/ggal_1_48850000_49020000.Ggal71.500bpflank.fa -i index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (mapping-based) v1.0.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ threads ] => { 1 }\n",
      "### [ libType ] => { U }\n",
      "### [ index ] => { index }\n",
      "### [ mates1 ] => { ./data/ggal/ggal_gut_1.fq }\n",
      "### [ mates2 ] => { ./data/ggal/ggal_gut_2.fq }\n",
      "### [ output ] => { results_salmon }\n",
      "Logs will be written to results_salmon/logs\n",
      "\u001b[00m[2020-01-17 16:47:29.930] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:29.930] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:29.930] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:29.930] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:29.930] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 11.652 us\n",
      "-----------------------------------------\n",
      "size = 14\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 18.315 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 4.739 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 52.308 us\n",
      "-----------------------------------------\n",
      "size = 171333\n",
      "Number of ones: 13\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 1\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 248.26 us\n",
      "-----------------------------------------\n",
      "size = 171333\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 27.144 us\n",
      "-----------------------------------------\n",
      "size = 170943\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 63.44 us\n",
      "-----------------------------------------\n",
      "size = 171001\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 23.572 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 5.067 us\n",
      "-----------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2020-01-17 16:47:29.972] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:29.973] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:29.973] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:29.973] [jointLog] [info] Index contained 1 targets\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:29.973] [jointLog] [info] Number of decoys : 0\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:29.973] [jointLog] [info] First decoy index : 18,446,744,073,709,551,615 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2020-01-17 16:47:30.015] [jointLog] [info] Computed 1 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.015] [jointLog] [info] Counted 2,058 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.016] [jointLog] [info] Number of mappings discarded because of alignment score : 1,540\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.016] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 874\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.016] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 0\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.016] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 1,037\n",
      "\u001b[00m\u001b[33m\u001b[1m[2020-01-17 16:47:30.016] [jointLog] [warning] Only 2058 fragments were mapped, but the number of burn-in fragments was set to 5000000.\n",
      "The effective lengths have been computed using the observed mappings.\n",
      "\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.016] [jointLog] [info] Mapping rate = 70.0715%\n",
      "\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.016] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.016] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.016] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.016] [jointLog] [info] iteration = 0 | max rel diff. = 2.20966e-16\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.016] [jointLog] [info] iteration = 100 | max rel diff. = 0\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.016] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2020-01-17 16:47:30.016] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00m\u001b[33m\u001b[1m[2020-01-17 16:47:30.016] [jointLog] [warning] NOTE: Read Lib [[ ./data/ggal/ggal_gut_1.fq, ./data/ggal/ggal_gut_2.fq]] :\n",
      "\n",
      "Detected a *potential* strand bias > 1% in an unstranded protocol check the file: results_salmon/lib_format_counts.json for details\n",
      "\n",
      "\u001b[00m"
     ]
    }
   ],
   "source": [
    "!salmon quant --threads 1 --libType=U -i index -1 ./data/ggal/ggal_gut_1.fq -2 ./data/ggal/ggal_gut_2.fq  -o results_salmon\n",
    "!rm -rf index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.exists(\"results_fastqc\"):\n",
    "    os.makedirs(\"results_fastqc\")\n",
    "    \n",
    "!fastqc -o results_fastqc -f fastq -q ./data/ggal/ggal_gut_*.fq\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/multiqc/utils/config.py:45: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  configs = yaml.load(f)\n",
      "/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/multiqc/utils/config.py:51: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  sp = yaml.load(f)\n",
      "[2020-01-17 16:47:41,548] multiqc                                            [DEBUG  ]  No MultiQC config found: /home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/multiqc_config.yaml\n",
      "[2020-01-17 16:47:41,548] multiqc                                            [DEBUG  ]  No MultiQC config found: /home/jespinosa/.multiqc_config.yaml\n",
      "/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/multiqc/utils/config.py:121: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  new_config = yaml.load(f)\n",
      "[2020-01-17 16:47:41,549] multiqc                                            [DEBUG  ]  Loading config settings from: multiqc_config.yaml\n",
      "[2020-01-17 16:47:41,549] multiqc                                            [DEBUG  ]  New config 'title': BovReg template for reproducible analysis\n",
      "[2020-01-17 16:47:41,549] multiqc                                            [DEBUG  ]  New config 'subtitle': Nextflow RNA-seq demo - A pipeline for quantification of genomic features from short read data\n",
      "[2020-01-17 16:47:41,549] multiqc                                            [DEBUG  ]  New config 'report_header_info': [{'Contact E-mail': 'Bovreg-wp3@ebi.ac.uk'}, {'Application Type': 'RNA-seq'}, {'Project Type': 'BovReg WP3 template'}]\n",
      "[2020-01-17 16:47:41,550] multiqc                                            [DEBUG  ]  New config 'custom_logo_url':  https://www.bovreg.eu\n",
      "[2020-01-17 16:47:41,550] multiqc                                            [DEBUG  ]  New config 'custom_logo_title': BovReg\n",
      "[2020-01-17 16:47:41,550] multiqc                                            [DEBUG  ]  New config 'custom_logo': /home/jespinosa/git/BovReg-Reproducibility/rnaseq-nf/multiqc_in/logo.png\n",
      "[2020-01-17 16:47:41,550] multiqc                                            [DEBUG  ]  Command used: /home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/bin/multiqc -v . --outdir ../results_multiqc/\n",
      "[2020-01-17 16:47:41,948] multiqc                                            [WARNING]  MultiQC Version v1.8 now available!\n",
      "[2020-01-17 16:47:41,948] multiqc                                            [INFO   ]  This is MultiQC v1.5.dev0\n",
      "[2020-01-17 16:47:41,948] multiqc                                            [DEBUG  ]  Command     : /home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/bin/multiqc -v . --outdir ../results_multiqc/\n",
      "[2020-01-17 16:47:41,948] multiqc                                            [DEBUG  ]  Working dir : /home/jespinosa/git/BovReg-Reproducibility/rnaseq-nf/multiqc_in\n",
      "[2020-01-17 16:47:41,949] multiqc                                            [INFO   ]  Template    : default\n",
      "[2020-01-17 16:47:41,949] multiqc                                            [INFO   ]  Report title: BovReg template for reproducible analysis\n",
      "[2020-01-17 16:47:41,949] multiqc                                            [INFO   ]  Searching '.'\n",
      "[2020-01-17 16:47:41,950] multiqc                                            [DEBUG  ]  Analysing modules: afterqc, custom_content, conpair, peddy, methylQA, qualimap, preseq, quast, qorts, rna_seqc, rsem, rseqc, busco, goleft_indexcov, disambiguate, supernova, deeptools, sargasso, verifybamid, homer, macs2, theta2, snpeff, gatk, htseq, bcftools, featureCounts, picard, prokka, samblaster, samtools, bamtools, jellyfish, vcftools, bbmap, bismark, hicexplorer, hicup, salmon, kallisto, slamdunk, star, hisat2, tophat, bowtie2, bowtie1, leehom, adapterRemoval, cutadapt, flexbar, trimmomatic, skewer, sortmerna, biobloomtools, fastq_screen, fastqc, clusterflow, bcl2fastq, interop\n",
      "[2020-01-17 16:47:41,950] multiqc                                            [DEBUG  ]  Using temporary directory for creating report: /tmp/tmpg968djww\n",
      "[2020-01-17 16:47:42,039] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: ggal_gut_1_fastqc.html\n",
      "[2020-01-17 16:47:42,039] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: ggal_gut_2_fastqc.html\n",
      "[2020-01-17 16:47:42,192] multiqc                                            [DEBUG  ]  No samples found: afterqc\n",
      "[2020-01-17 16:47:42,194] multiqc                                            [DEBUG  ]  No samples found: custom_content\n",
      "[2020-01-17 16:47:42,196] multiqc                                            [DEBUG  ]  No samples found: conpair\n",
      "[2020-01-17 16:47:42,198] multiqc                                            [DEBUG  ]  No samples found: peddy\n",
      "[2020-01-17 16:47:42,200] multiqc                                            [DEBUG  ]  No samples found: methylQA\n",
      "[2020-01-17 16:47:42,202] multiqc                                            [DEBUG  ]  No samples found: qualimap\n",
      "[2020-01-17 16:47:42,203] multiqc                                            [DEBUG  ]  No samples found: preseq\n",
      "[2020-01-17 16:47:42,205] multiqc                                            [DEBUG  ]  No samples found: quast\n",
      "[2020-01-17 16:47:42,208] multiqc                                            [DEBUG  ]  No samples found: qorts\n",
      "[2020-01-17 16:47:42,210] multiqc                                            [DEBUG  ]  No samples found: rna_seqc\n",
      "[2020-01-17 16:47:42,212] multiqc                                            [DEBUG  ]  No samples found: rsem\n",
      "[2020-01-17 16:47:42,215] multiqc                                            [DEBUG  ]  No samples found: rseqc\n",
      "[2020-01-17 16:47:42,216] multiqc                                            [DEBUG  ]  No samples found: busco\n",
      "[2020-01-17 16:47:42,218] multiqc                                            [DEBUG  ]  No samples found: goleft_indexcov\n",
      "[2020-01-17 16:47:42,220] multiqc                                            [DEBUG  ]  No samples found: disambiguate\n",
      "[2020-01-17 16:47:42,222] multiqc                                            [DEBUG  ]  No samples found: supernova\n",
      "[2020-01-17 16:47:42,224] multiqc                                            [DEBUG  ]  No samples found: deeptools\n",
      "[2020-01-17 16:47:42,226] multiqc                                            [DEBUG  ]  No samples found: sargasso\n",
      "[2020-01-17 16:47:42,228] multiqc                                            [DEBUG  ]  No samples found: verifybamid\n",
      "[2020-01-17 16:47:42,230] multiqc                                            [DEBUG  ]  No samples found: homer\n",
      "[2020-01-17 16:47:42,232] multiqc                                            [DEBUG  ]  No samples found: macs2\n",
      "[2020-01-17 16:47:42,233] multiqc                                            [DEBUG  ]  No samples found: theta2\n",
      "[2020-01-17 16:47:42,235] multiqc                                            [DEBUG  ]  No samples found: snpeff\n",
      "[2020-01-17 16:47:42,237] multiqc                                            [DEBUG  ]  No samples found: gatk\n",
      "[2020-01-17 16:47:42,239] multiqc                                            [DEBUG  ]  No samples found: htseq\n",
      "[2020-01-17 16:47:42,241] multiqc                                            [DEBUG  ]  No samples found: bcftools\n",
      "[2020-01-17 16:47:42,243] multiqc                                            [DEBUG  ]  No samples found: featureCounts\n",
      "[2020-01-17 16:47:42,246] multiqc                                            [DEBUG  ]  No samples found: picard\n",
      "[2020-01-17 16:47:42,248] multiqc                                            [DEBUG  ]  No samples found: prokka\n",
      "[2020-01-17 16:47:42,250] multiqc                                            [DEBUG  ]  No samples found: samblaster\n",
      "[2020-01-17 16:47:42,252] multiqc                                            [DEBUG  ]  No samples found: samtools\n",
      "[2020-01-17 16:47:42,254] multiqc                                            [DEBUG  ]  No samples found: bamtools\n",
      "[2020-01-17 16:47:42,256] multiqc                                            [DEBUG  ]  No samples found: jellyfish\n",
      "[2020-01-17 16:47:42,258] multiqc                                            [DEBUG  ]  No samples found: vcftools\n",
      "[2020-01-17 16:47:42,261] multiqc                                            [DEBUG  ]  No samples found: bbmap\n",
      "[2020-01-17 16:47:42,263] multiqc                                            [DEBUG  ]  No samples found: bismark\n",
      "[2020-01-17 16:47:42,265] multiqc                                            [DEBUG  ]  No samples found: hicexplorer\n",
      "[2020-01-17 16:47:42,267] multiqc                                            [DEBUG  ]  No samples found: hicup\n",
      "[2020-01-17 16:47:42,270] multiqc.modules.salmon.salmon                      [INFO   ]  Found 1 meta reports\n",
      "[2020-01-17 16:47:42,270] multiqc.modules.salmon.salmon                      [INFO   ]  Found 1 fragment length distributions\n",
      "[2020-01-17 16:47:42,274] multiqc                                            [DEBUG  ]  No samples found: kallisto\n",
      "[2020-01-17 16:47:42,275] multiqc                                            [DEBUG  ]  No samples found: slamdunk\n",
      "[2020-01-17 16:47:42,277] multiqc                                            [DEBUG  ]  No samples found: star\n",
      "[2020-01-17 16:47:42,279] multiqc                                            [DEBUG  ]  No samples found: hisat2\n",
      "[2020-01-17 16:47:42,281] multiqc                                            [DEBUG  ]  No samples found: tophat\n",
      "[2020-01-17 16:47:42,282] multiqc                                            [DEBUG  ]  No samples found: bowtie2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2020-01-17 16:47:42,284] multiqc                                            [DEBUG  ]  No samples found: bowtie1\n",
      "[2020-01-17 16:47:42,286] multiqc                                            [DEBUG  ]  No samples found: leehom\n",
      "[2020-01-17 16:47:42,288] multiqc                                            [DEBUG  ]  No samples found: adapterRemoval\n",
      "[2020-01-17 16:47:42,290] multiqc                                            [DEBUG  ]  No samples found: cutadapt\n",
      "[2020-01-17 16:47:42,292] multiqc                                            [DEBUG  ]  No samples found: flexbar\n",
      "[2020-01-17 16:47:42,293] multiqc                                            [DEBUG  ]  No samples found: trimmomatic\n",
      "[2020-01-17 16:47:42,295] multiqc                                            [DEBUG  ]  No samples found: skewer\n",
      "[2020-01-17 16:47:42,297] multiqc                                            [DEBUG  ]  No samples found: sortmerna\n",
      "[2020-01-17 16:47:42,299] multiqc                                            [DEBUG  ]  No samples found: biobloomtools\n",
      "[2020-01-17 16:47:42,301] multiqc                                            [DEBUG  ]  No samples found: fastq_screen\n",
      "[2020-01-17 16:47:42,306] multiqc.modules.fastqc.fastqc                      [INFO   ]  Found 2 reports\n",
      "[2020-01-17 16:47:42,331] multiqc                                            [DEBUG  ]  No samples found: clusterflow\n",
      "[2020-01-17 16:47:42,332] multiqc                                            [DEBUG  ]  No samples found: bcl2fastq\n",
      "[2020-01-17 16:47:42,334] multiqc                                            [DEBUG  ]  No samples found: interop\n",
      "[2020-01-17 16:47:42,337] multiqc                                            [INFO   ]  Compressing plot data\n",
      "[2020-01-17 16:47:42,365] multiqc                                            [WARNING]  Previous MultiQC output found! Adjusting filenames..\n",
      "[2020-01-17 16:47:42,365] multiqc                                            [WARNING]  Use -f or --force to overwrite existing reports instead\n",
      "[2020-01-17 16:47:42,365] multiqc                                            [INFO   ]  Report      : ../results_multiqc/multiqc_report_2.html\n",
      "[2020-01-17 16:47:42,365] multiqc                                            [INFO   ]  Data        : ../results_multiqc/multiqc_data_2\n",
      "[2020-01-17 16:47:42,365] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpg968djww/multiqc_data/multiqc_salmon.txt' to '../results_multiqc/multiqc_data_2'\n",
      "[2020-01-17 16:47:42,365] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpg968djww/multiqc_data/multiqc_fastqc.txt' to '../results_multiqc/multiqc_data_2'\n",
      "[2020-01-17 16:47:42,365] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpg968djww/multiqc_data/multiqc_sources.txt' to '../results_multiqc/multiqc_data_2'\n",
      "[2020-01-17 16:47:42,365] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpg968djww/multiqc_data/multiqc_data.json' to '../results_multiqc/multiqc_data_2'\n",
      "[2020-01-17 16:47:42,365] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpg968djww/multiqc_data/multiqc_general_stats.txt' to '../results_multiqc/multiqc_data_2'\n",
      "[2020-01-17 16:47:42,428] multiqc                                            [INFO   ]  MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "if not os.path.exists(\"multiqc_in\"):\n",
    "    os.makedirs(\"multiqc_in\")\n",
    "\n",
    "if not os.path.exists(\"results_multiqc\"):\n",
    "    os.makedirs(\"results_multiqc\")\n",
    "\n",
    "!cp results_fastqc/* multiqc_in\n",
    "!cp -rf results_salmon/* multiqc_in\n",
    "!cp multiqc/* multiqc_in\n",
    "!echo \"custom_logo: $PWD/multiqc_in/logo.png\" >> multiqc_in/multiqc_config.yaml\n",
    "\n",
    "pwd = os.getcwd()\n",
    "work_dir = pwd + \"/multiqc_in\"\n",
    "os.chdir(work_dir)\n",
    "os.getcwd()\n",
    "\n",
    "!multiqc -v . --outdir ../results_multiqc/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/IPython/core/interactiveshell.py\", line 3319, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-7-8c01a606a94a>\", line 1, in <module>\n",
      "    shutil.rmtree(work_dir)\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/shutil.py\", line 477, in rmtree\n",
      "    onerror(os.lstat, path, sys.exc_info())\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/shutil.py\", line 475, in rmtree\n",
      "    orig_st = os.lstat(path)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '/home/jespinosa/git/BovReg-Reproducibility/rnaseq-nf/multiqc_in'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/IPython/core/interactiveshell.py\", line 2034, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'FileNotFoundError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/IPython/core/ultratb.py\", line 1151, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/IPython/core/ultratb.py\", line 319, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/IPython/core/ultratb.py\", line 353, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/inspect.py\", line 725, in getmodule\n",
      "    file = getabsfile(object, _filename)\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/inspect.py\", line 709, in getabsfile\n",
      "    return os.path.normcase(os.path.abspath(_filename))\n",
      "  File \"/home/jespinosa/conda_env/nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/posixpath.py\", line 383, in abspath\n",
      "    cwd = os.getcwd()\n",
      "FileNotFoundError: [Errno 2] No such file or directory\n"
     ]
    },
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/home/jespinosa/git/BovReg-Reproducibility/rnaseq-nf/multiqc_in'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m"
     ]
    }
   ],
   "source": [
    "shutil.rmtree(work_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "work_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
