{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run pipeline with bash \n",
    "\n",
    "You will need to install Salmon, fastQC and multiQC before running the commands below. Otherwise, you can create the conda environment, activate it and run the notebook then, as explained [here](create_conda_env.ipynb). \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os, shutil\n",
    "pwd = os.getcwd()\n",
    "os.getcwd()\n",
    "pipeline_folder = \"rnaseq-nf\" \n",
    "work_dir = pwd + \"/../\" + pipeline_folder\n",
    "os.chdir(work_dir)\n",
    "# os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "index [\"index\"] did not previously exist  . . . creating it\n",
      "[2020-01-28 08:30:06.876] [jLog] [info] building index\n",
      "out : index\n",
      "\u001b[00m[2020-01-28 08:30:06.876] [puff::index::jointLog] [info] Running fixFasta\n",
      "\u001b[00m\n",
      "[Step 1 of 4] : counting k-mers\n",
      "\n",
      "\u001b[00m[2020-01-28 08:30:06.895] [puff::index::jointLog] [info] Replaced 0 non-ATCG nucleotides\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:06.895] [puff::index::jointLog] [info] Clipped poly-A tails from 0 transcripts\n",
      "\u001b[00mwrote 1 cleaned references\n",
      "seqHash 256 : c7b06752fa0748ecfa05ab97e8460b69c408b6d5fedf9de3d0d53b05570cf942\n",
      "seqHash 512 : ed859a805ddc0e6805be16dea7160060c1ecf1853f01fe24e64779390dd0b4aa157dab4d3e6234e9f7a1a80cd74df05454259b325624f601a0f86ea508550a8e\n",
      "nameHash 256 : 6b86b273ff34fce19d6b804eff5a3f5747ada4eaa22f1d49c01e52ddb7875b4b\n",
      "nameHash 512 : 4dff4ea340f0a823f15d3f4f01ab62eae0e5da579ccb851f8db9dfe84c58b2b37b89903a740e1ee172da793a6e79d560e5f7f9bd058a12a280433ed6fa46510a\n",
      "\u001b[00m[2020-01-28 08:30:06.897] [puff::index::jointLog] [info] Filter size not provided; estimating from number of distinct k-mers\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:06.908] [puff::index::jointLog] [info] ntHll estimated 147846 distinct k-mers, setting filter size to 2^22\n",
      "\u001b[00mThreads = 1\n",
      "Vertex length = 31\n",
      "Hash functions = 5\n",
      "Filter size = 4194304\n",
      "Capacity = 2\n",
      "Files: \n",
      "index/ref_k31_fixed.fa\n",
      "--------------------------------------------------------------------------------\n",
      "Round 0, 0:4194304\n",
      "Pass\tFilling\tFiltering\n",
      "1\t0\t0\t\n",
      "2\t0\t0\n",
      "True junctions count = 7\n",
      "False junctions count = 246\n",
      "Hash table size = 253\n",
      "Candidate marks count = 260\n",
      "--------------------------------------------------------------------------------\n",
      "Reallocating bifurcations time: 0\n",
      "True marks count: 16\n",
      "Edges construction time: 0\n",
      "--------------------------------------------------------------------------------\n",
      "Distinct junctions = 7\n",
      "\n",
      "approximateContigTotalLength: 92130\n",
      "counters:\n",
      "1 0 0 0\n",
      "contig count: 14 element count: 171333 complex nodes: 1\n",
      "size: 171333\n",
      "# of ones in rank vector: 13\n",
      "size: 171333\n",
      "\u001b[00m[2020-01-28 08:30:07.010] [puff::index::jointLog] [info] Setting the index/BinaryGfa directory index\n",
      "\u001b[00msize = 171333\n",
      "-----------------------------------------\n",
      "| Loading contigs | Time = 21.323 us\n",
      "-----------------------------------------\n",
      "size = 171333\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 8.418 us\n",
      "-----------------------------------------\n",
      "Number of ones: 13\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 1\n",
      "\u001b[00m[2020-01-28 08:30:07.011] [puff::index::jointLog] [info] Done wrapping the rank vector with a rank9sel structure.\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.011] [puff::index::jointLog] [info] contig count for validation: 13\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.011] [puff::index::jointLog] [info] Total # of Contigs : 13\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.011] [puff::index::jointLog] [info] Total # of numerical Contigs : 13\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.011] [puff::index::jointLog] [info] \n",
      "Total # of segments we have position for : 13\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.011] [puff::index::jointLog] [info] total contig vec entries 17\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.011] [puff::index::jointLog] [info] bits per offset entry 5\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.011] [puff::index::jointLog] [info] there were 1  equivalence classes\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.011] [puff::index::jointLog] [info] # segments = 13\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.011] [puff::index::jointLog] [info] total length = 171,333\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.011] [puff::index::jointLog] [info] Reading the reference files ...\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.012] [puff::index::jointLog] [info] positional integer width = 18\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.012] [puff::index::jointLog] [info] seqSize = 171,333\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.012] [puff::index::jointLog] [info] rankSize = 171,333\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.012] [puff::index::jointLog] [info] edgeVecSize = 0\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.012] [puff::index::jointLog] [info] num keys = 170,943\n",
      "\u001b[00mfor info, total work write each  : 2.331    total work inram from level 3 : 4.322  total work raw : 25.000 \n",
      "[Building BooPHF]  100  %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec\n",
      "Bitarray          901696  bits (100.00 %)   (array + ranks )\n",
      "final hash             0  bits (0.00 %) (nb in final hash 0)\n",
      "\u001b[00m[2020-01-28 08:30:07.025] [puff::index::jointLog] [info] mphf size = 0.107491 MB\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.025] [puff::index::jointLog] [info] chunk size = 171,333\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.025] [puff::index::jointLog] [info] chunk 0 = [0, 171,303)\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.034] [puff::index::jointLog] [info] finished populating pos vector\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.034] [puff::index::jointLog] [info] writing index components\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.035] [puff::index::jointLog] [info] finished writing dense pufferfish index\n",
      "\u001b[00m[2020-01-28 08:30:07.035] [jLog] [info] done building index\n"
     ]
    }
   ],
   "source": [
    "!salmon index --threads 1 -t ./data/ggal/ggal_1_48850000_49020000.Ggal71.500bpflank.fa -i index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (mapping-based) v1.0.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ threads ] => { 1 }\n",
      "### [ libType ] => { U }\n",
      "### [ index ] => { index }\n",
      "### [ mates1 ] => { ./data/ggal/ggal_gut_1.fq }\n",
      "### [ mates2 ] => { ./data/ggal/ggal_gut_2.fq }\n",
      "### [ output ] => { results_salmon }\n",
      "Logs will be written to results_salmon/logs\n",
      "\u001b[00m[2020-01-28 08:30:07.460] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.460] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.460] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.460] [jointLog] [info] parsing read library format\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.461] [jointLog] [info] There is 1 library.\n",
      "\u001b[00m-----------------------------------------\n",
      "| Loading contig table | Time = 10.816 us\n",
      "-----------------------------------------\n",
      "size = 14\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 14.332 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 3.891 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 64.634 us\n",
      "-----------------------------------------\n",
      "size = 171333\n",
      "Number of ones: 13\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 1\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 233.89 us\n",
      "-----------------------------------------\n",
      "size = 171333\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 23.614 us\n",
      "-----------------------------------------\n",
      "size = 170943\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 67.888 us\n",
      "-----------------------------------------\n",
      "size = 171001\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 22.83 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 4.393 us\n",
      "-----------------------------------------\n",
      "\u001b[00m[2020-01-28 08:30:07.501] [jointLog] [info] Loading pufferfish index\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.501] [jointLog] [info] Loading dense pufferfish index.\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.502] [jointLog] [info] done\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.502] [jointLog] [info] Index contained 1 targets\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.502] [jointLog] [info] Number of decoys : 0\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.502] [jointLog] [info] First decoy index : 18,446,744,073,709,551,615 \n",
      "\u001b[00m\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[00m[2020-01-28 08:30:07.558] [jointLog] [info] Computed 1 rich equivalence classes for further processing\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.558] [jointLog] [info] Counted 2,058 total reads in the equivalence classes \n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.559] [jointLog] [info] Number of mappings discarded because of alignment score : 1,540\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.559] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 874\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.559] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 0\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.559] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 1,037\n",
      "\u001b[00m\u001b[33m\u001b[1m[2020-01-28 08:30:07.559] [jointLog] [warning] Only 2058 fragments were mapped, but the number of burn-in fragments was set to 5000000.\n",
      "The effective lengths have been computed using the observed mappings.\n",
      "\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.559] [jointLog] [info] Mapping rate = 70.0715%\n",
      "\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.559] [jointLog] [info] finished quantifyLibrary()\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.559] [jointLog] [info] Starting optimizer\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.560] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.560] [jointLog] [info] iteration = 0 | max rel diff. = 2.20966e-16\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.560] [jointLog] [info] iteration = 100 | max rel diff. = 0\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.560] [jointLog] [info] Finished optimizer\n",
      "\u001b[00m\u001b[00m[2020-01-28 08:30:07.560] [jointLog] [info] writing output \n",
      "\n",
      "\u001b[00m\u001b[33m\u001b[1m[2020-01-28 08:30:07.560] [jointLog] [warning] NOTE: Read Lib [[ ./data/ggal/ggal_gut_1.fq, ./data/ggal/ggal_gut_2.fq]] :\n",
      "\n",
      "Detected a *potential* strand bias > 1% in an unstranded protocol check the file: results_salmon/lib_format_counts.json for details\n",
      "\n",
      "\u001b[00m"
     ]
    }
   ],
   "source": [
    "!salmon quant --threads 1 --libType=U -i index -1 ./data/ggal/ggal_gut_1.fq -2 ./data/ggal/ggal_gut_2.fq  -o results_salmon\n",
    "!rm -rf index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.exists(\"results_fastqc\"):\n",
    "    os.makedirs(\"results_fastqc\")\n",
    "    \n",
    "!fastqc -o results_fastqc -f fastq -q ./data/ggal/ggal_gut_*.fq\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/kadomu/conda_env_nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/multiqc/utils/config.py:45: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  configs = yaml.load(f)\n",
      "/home/kadomu/conda_env_nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/multiqc/utils/config.py:51: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  sp = yaml.load(f)\n",
      "[2020-01-28 08:30:13,422] multiqc                                            [DEBUG  ]  No MultiQC config found: /home/kadomu/conda_env_nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/multiqc_config.yaml\n",
      "[2020-01-28 08:30:13,422] multiqc                                            [DEBUG  ]  No MultiQC config found: /home/kadomu/.multiqc_config.yaml\n",
      "/home/kadomu/conda_env_nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/lib/python3.6/site-packages/multiqc/utils/config.py:121: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.\n",
      "  new_config = yaml.load(f)\n",
      "[2020-01-28 08:30:13,423] multiqc                                            [DEBUG  ]  Loading config settings from: multiqc_config.yaml\n",
      "[2020-01-28 08:30:13,423] multiqc                                            [DEBUG  ]  New config 'title': BovReg template for reproducible analysis\n",
      "[2020-01-28 08:30:13,424] multiqc                                            [DEBUG  ]  New config 'subtitle': Nextflow RNA-seq demo - A pipeline for quantification of genomic features from short read data\n",
      "[2020-01-28 08:30:13,424] multiqc                                            [DEBUG  ]  New config 'report_header_info': [{'Contact E-mail': 'Bovreg-wp3@ebi.ac.uk'}, {'Application Type': 'RNA-seq'}, {'Project Type': 'BovReg WP3 template'}]\n",
      "[2020-01-28 08:30:13,424] multiqc                                            [DEBUG  ]  New config 'custom_logo_url':  https://www.bovreg.eu\n",
      "[2020-01-28 08:30:13,424] multiqc                                            [DEBUG  ]  New config 'custom_logo_title': BovReg\n",
      "[2020-01-28 08:30:13,424] multiqc                                            [DEBUG  ]  New config 'custom_logo': /home/kadomu/git/BovReg-Reproducibility/rnaseq-nf/multiqc_in/logo.png\n",
      "[2020-01-28 08:30:13,424] multiqc                                            [DEBUG  ]  Command used: /home/kadomu/conda_env_nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/bin/multiqc -v . --outdir ../results_multiqc/\n",
      "[2020-01-28 08:30:14,562] multiqc                                            [WARNING]  MultiQC Version v1.8 now available!\n",
      "[2020-01-28 08:30:14,563] multiqc                                            [INFO   ]  This is MultiQC v1.5.dev0\n",
      "[2020-01-28 08:30:14,563] multiqc                                            [DEBUG  ]  Command     : /home/kadomu/conda_env_nxf/rnaseq-nf-58fed063cfbc6adc6b42861c7e7fae6d/bin/multiqc -v . --outdir ../results_multiqc/\n",
      "[2020-01-28 08:30:14,564] multiqc                                            [DEBUG  ]  Working dir : /home/kadomu/git/BovReg-Reproducibility/rnaseq-nf/multiqc_in\n",
      "[2020-01-28 08:30:14,564] multiqc                                            [INFO   ]  Template    : default\n",
      "[2020-01-28 08:30:14,565] multiqc                                            [INFO   ]  Report title: BovReg template for reproducible analysis\n",
      "[2020-01-28 08:30:14,565] multiqc                                            [INFO   ]  Searching '.'\n",
      "[2020-01-28 08:30:14,567] multiqc                                            [DEBUG  ]  Analysing modules: afterqc, custom_content, conpair, peddy, methylQA, qualimap, preseq, quast, qorts, rna_seqc, rsem, rseqc, busco, goleft_indexcov, disambiguate, supernova, deeptools, sargasso, verifybamid, homer, macs2, theta2, snpeff, gatk, htseq, bcftools, featureCounts, picard, prokka, samblaster, samtools, bamtools, jellyfish, vcftools, bbmap, bismark, hicexplorer, hicup, salmon, kallisto, slamdunk, star, hisat2, tophat, bowtie2, bowtie1, leehom, adapterRemoval, cutadapt, flexbar, trimmomatic, skewer, sortmerna, biobloomtools, fastq_screen, fastqc, clusterflow, bcl2fastq, interop\n",
      "[2020-01-28 08:30:14,568] multiqc                                            [DEBUG  ]  Using temporary directory for creating report: /tmp/tmpph_7r93p\n",
      "[2020-01-28 08:30:14,666] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: ggal_gut_1_fastqc.html\n",
      "[2020-01-28 08:30:14,688] multiqc                                            [DEBUG  ]  Ignoring file as matched an ignore pattern: ggal_gut_2_fastqc.html\n",
      "[2020-01-28 08:30:14,907] multiqc                                            [DEBUG  ]  No samples found: afterqc\n",
      "[2020-01-28 08:30:14,910] multiqc                                            [DEBUG  ]  No samples found: custom_content\n",
      "[2020-01-28 08:30:14,912] multiqc                                            [DEBUG  ]  No samples found: conpair\n",
      "[2020-01-28 08:30:14,914] multiqc                                            [DEBUG  ]  No samples found: peddy\n",
      "[2020-01-28 08:30:14,916] multiqc                                            [DEBUG  ]  No samples found: methylQA\n",
      "[2020-01-28 08:30:14,918] multiqc                                            [DEBUG  ]  No samples found: qualimap\n",
      "[2020-01-28 08:30:14,920] multiqc                                            [DEBUG  ]  No samples found: preseq\n",
      "[2020-01-28 08:30:14,922] multiqc                                            [DEBUG  ]  No samples found: quast\n",
      "[2020-01-28 08:30:14,925] multiqc                                            [DEBUG  ]  No samples found: qorts\n",
      "[2020-01-28 08:30:14,927] multiqc                                            [DEBUG  ]  No samples found: rna_seqc\n",
      "[2020-01-28 08:30:14,929] multiqc                                            [DEBUG  ]  No samples found: rsem\n",
      "[2020-01-28 08:30:14,932] multiqc                                            [DEBUG  ]  No samples found: rseqc\n",
      "[2020-01-28 08:30:14,934] multiqc                                            [DEBUG  ]  No samples found: busco\n",
      "[2020-01-28 08:30:14,936] multiqc                                            [DEBUG  ]  No samples found: goleft_indexcov\n",
      "[2020-01-28 08:30:14,938] multiqc                                            [DEBUG  ]  No samples found: disambiguate\n",
      "[2020-01-28 08:30:14,940] multiqc                                            [DEBUG  ]  No samples found: supernova\n",
      "[2020-01-28 08:30:14,942] multiqc                                            [DEBUG  ]  No samples found: deeptools\n",
      "[2020-01-28 08:30:14,945] multiqc                                            [DEBUG  ]  No samples found: sargasso\n",
      "[2020-01-28 08:30:14,947] multiqc                                            [DEBUG  ]  No samples found: verifybamid\n",
      "[2020-01-28 08:30:14,950] multiqc                                            [DEBUG  ]  No samples found: homer\n",
      "[2020-01-28 08:30:14,952] multiqc                                            [DEBUG  ]  No samples found: macs2\n",
      "[2020-01-28 08:30:14,955] multiqc                                            [DEBUG  ]  No samples found: theta2\n",
      "[2020-01-28 08:30:14,957] multiqc                                            [DEBUG  ]  No samples found: snpeff\n",
      "[2020-01-28 08:30:14,960] multiqc                                            [DEBUG  ]  No samples found: gatk\n",
      "[2020-01-28 08:30:14,963] multiqc                                            [DEBUG  ]  No samples found: htseq\n",
      "[2020-01-28 08:30:14,965] multiqc                                            [DEBUG  ]  No samples found: bcftools\n",
      "[2020-01-28 08:30:14,967] multiqc                                            [DEBUG  ]  No samples found: featureCounts\n",
      "[2020-01-28 08:30:14,970] multiqc                                            [DEBUG  ]  No samples found: picard\n",
      "[2020-01-28 08:30:14,972] multiqc                                            [DEBUG  ]  No samples found: prokka\n",
      "[2020-01-28 08:30:14,974] multiqc                                            [DEBUG  ]  No samples found: samblaster\n",
      "[2020-01-28 08:30:14,976] multiqc                                            [DEBUG  ]  No samples found: samtools\n",
      "[2020-01-28 08:30:14,978] multiqc                                            [DEBUG  ]  No samples found: bamtools\n",
      "[2020-01-28 08:30:14,980] multiqc                                            [DEBUG  ]  No samples found: jellyfish\n",
      "[2020-01-28 08:30:14,983] multiqc                                            [DEBUG  ]  No samples found: vcftools\n",
      "[2020-01-28 08:30:14,986] multiqc                                            [DEBUG  ]  No samples found: bbmap\n",
      "[2020-01-28 08:30:14,988] multiqc                                            [DEBUG  ]  No samples found: bismark\n",
      "[2020-01-28 08:30:14,990] multiqc                                            [DEBUG  ]  No samples found: hicexplorer\n",
      "[2020-01-28 08:30:14,992] multiqc                                            [DEBUG  ]  No samples found: hicup\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2020-01-28 08:30:14,996] multiqc.modules.salmon.salmon                      [INFO   ]  Found 1 meta reports\n",
      "[2020-01-28 08:30:14,996] multiqc.modules.salmon.salmon                      [INFO   ]  Found 1 fragment length distributions\n",
      "[2020-01-28 08:30:15,000] multiqc                                            [DEBUG  ]  No samples found: kallisto\n",
      "[2020-01-28 08:30:15,003] multiqc                                            [DEBUG  ]  No samples found: slamdunk\n",
      "[2020-01-28 08:30:15,006] multiqc                                            [DEBUG  ]  No samples found: star\n",
      "[2020-01-28 08:30:15,008] multiqc                                            [DEBUG  ]  No samples found: hisat2\n",
      "[2020-01-28 08:30:15,010] multiqc                                            [DEBUG  ]  No samples found: tophat\n",
      "[2020-01-28 08:30:15,013] multiqc                                            [DEBUG  ]  No samples found: bowtie2\n",
      "[2020-01-28 08:30:15,015] multiqc                                            [DEBUG  ]  No samples found: bowtie1\n",
      "[2020-01-28 08:30:15,017] multiqc                                            [DEBUG  ]  No samples found: leehom\n",
      "[2020-01-28 08:30:15,019] multiqc                                            [DEBUG  ]  No samples found: adapterRemoval\n",
      "[2020-01-28 08:30:15,021] multiqc                                            [DEBUG  ]  No samples found: cutadapt\n",
      "[2020-01-28 08:30:15,023] multiqc                                            [DEBUG  ]  No samples found: flexbar\n",
      "[2020-01-28 08:30:15,024] multiqc                                            [DEBUG  ]  No samples found: trimmomatic\n",
      "[2020-01-28 08:30:15,026] multiqc                                            [DEBUG  ]  No samples found: skewer\n",
      "[2020-01-28 08:30:15,028] multiqc                                            [DEBUG  ]  No samples found: sortmerna\n",
      "[2020-01-28 08:30:15,030] multiqc                                            [DEBUG  ]  No samples found: biobloomtools\n",
      "[2020-01-28 08:30:15,032] multiqc                                            [DEBUG  ]  No samples found: fastq_screen\n",
      "[2020-01-28 08:30:15,038] multiqc.modules.fastqc.fastqc                      [INFO   ]  Found 2 reports\n",
      "[2020-01-28 08:30:15,065] multiqc                                            [DEBUG  ]  No samples found: clusterflow\n",
      "[2020-01-28 08:30:15,067] multiqc                                            [DEBUG  ]  No samples found: bcl2fastq\n",
      "[2020-01-28 08:30:15,070] multiqc                                            [DEBUG  ]  No samples found: interop\n",
      "[2020-01-28 08:30:15,073] multiqc                                            [INFO   ]  Compressing plot data\n",
      "[2020-01-28 08:30:15,104] multiqc                                            [WARNING]  Previous MultiQC output found! Adjusting filenames..\n",
      "[2020-01-28 08:30:15,104] multiqc                                            [WARNING]  Use -f or --force to overwrite existing reports instead\n",
      "[2020-01-28 08:30:15,104] multiqc                                            [INFO   ]  Report      : ../results_multiqc/multiqc_report_4.html\n",
      "[2020-01-28 08:30:15,105] multiqc                                            [INFO   ]  Data        : ../results_multiqc/multiqc_data_4\n",
      "[2020-01-28 08:30:15,105] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpph_7r93p/multiqc_data/multiqc_fastqc.txt' to '../results_multiqc/multiqc_data_4'\n",
      "[2020-01-28 08:30:15,105] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpph_7r93p/multiqc_data/multiqc_sources.txt' to '../results_multiqc/multiqc_data_4'\n",
      "[2020-01-28 08:30:15,105] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpph_7r93p/multiqc_data/multiqc_general_stats.txt' to '../results_multiqc/multiqc_data_4'\n",
      "[2020-01-28 08:30:15,105] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpph_7r93p/multiqc_data/multiqc_salmon.txt' to '../results_multiqc/multiqc_data_4'\n",
      "[2020-01-28 08:30:15,105] multiqc                                            [DEBUG  ]  Moving data file from '/tmp/tmpph_7r93p/multiqc_data/multiqc_data.json' to '../results_multiqc/multiqc_data_4'\n",
      "[2020-01-28 08:30:15,168] multiqc                                            [INFO   ]  MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "if not os.path.exists(\"multiqc_in\"):\n",
    "    os.makedirs(\"multiqc_in\")\n",
    "\n",
    "if not os.path.exists(\"results_multiqc\"):\n",
    "    os.makedirs(\"results_multiqc\")\n",
    "\n",
    "!cp results_fastqc/* multiqc_in\n",
    "!cp -rf results_salmon/* multiqc_in\n",
    "!cp multiqc/* multiqc_in\n",
    "!echo \"custom_logo: $PWD/multiqc_in/logo.png\" >> multiqc_in/multiqc_config.yaml\n",
    "\n",
    "pwd = os.getcwd()\n",
    "work_dir = pwd + \"/multiqc_in\"\n",
    "os.chdir(work_dir)\n",
    "os.getcwd()\n",
    "\n",
    "!multiqc -v . --outdir ../results_multiqc/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "shutil.rmtree(work_dir)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
